### PR TITLE
chore(vendor): update go-common-libs/go-iscsi-utils and resolve breaking changes

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/7960

Update dependencis and resolve the breaking changes.
- go-iscsi-helper
- go-common-libs
The PR is mainly based on this PR: https://github.com/longhorn/longhorn-engine/pull/919/files#diff-0fa449830b5dc2b274b7c95a58b5071edc23f1e44af84a99c1fa9e600767829f

### CODE REVIEW

**Comment by ChanYiLin** [comment]
**CORE TEST SUCCEEDED**
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/6623/console

File: app/cmd/restore_to_file.go [hunk 1]
```diff
@@ -93,7 +94,7 @@ func restore(url string, concurrentLimit int) error {
 	bar := pb.StartNew(100)
 	periodicChecker := time.NewTicker(PeriodicRefreshIntervalInSeconds * time.Second)
 
-	progressKeyName := requestedBackupName + util.RandomID(8)
+	progressKeyName := requestedBackupName + util.RandomID()
```
Comment (derekbit) [inline app/cmd/restore_to_file.go:97]: Move `RandomID` function to go-common-libs.
nit: I'm fine with the parameter, randomIDLenth` which makes the function more generalized.

File: pkg/util/util.go [hunk 2]
```diff
@@ -316,3 +287,11 @@ func GetAddresses(volumeName, address string, dataServerProtocol types.DataServe
...
 	}
 }
+
+func UUID() string {
+	return uuid.New().String()
+}
+
+func RandomID() string {
+	return UUID()[:randomIDLenth]
+}
```
Comment (derekbit) [inline pkg/util/util.go:297]: Move to go-common-libs.

**Review by derekbit — COMMENTED** [review]
In general, LGTM.

File: app/cmd/restore_to_file.go (hunk 1 above)
Comment (ChanYiLin) [reply app/cmd/restore_to_file.go]: Fixed, thanks!

File: pkg/util/util.go (hunk 2 above)
Comment (ChanYiLin) [reply pkg/util/util.go]: Fixed, thanks

**Review by derekbit — DISMISSED** [review]
LGTM

File: app/cmd/restore_to_file.go [hunk 3]
```diff
@@ -13,7 +13,9 @@ import (
 	"github.com/urfave/cli"
 	"gopkg.in/cheggaaa/pb.v2"
 
-	iutil "github.com/longhorn/go-iscsi-helper/util"
+	lhexec "github.com/longhorn/go-common-libs/exec"
+	lhtypes "github.com/longhorn/go-common-libs/types"
+	commonUtils "github.com/longhorn/go-common-libs/utils"
```
Comment (innobead) [inline app/cmd/restore_to_file.go:18]: nit: suggest using the same prefix for the package alias. for example, here we can just use `commonExec`, `commonTypes` and `commonUtils`.

This minor consistent alignment can be improved later.

File: app/cmd/shutdown.go [hunk 4]
```diff
@@ -20,7 +20,7 @@ func addShutdown(f func() (err error)) {
 	}
 
 	hooks = append(hooks, f)
-	logrus.Debugf("Added shutdown func %v", util.GetFunctionName(f))
+	logrus.Debugf("Added shutdown func %v", lhutils.GetFunctionPath(f))
```
Comment (innobead) [inline app/cmd/shutdown.go:23]: Why not still use `lhutils.GetFunctionName`?

File: pkg/util/util.go [hunk 5]
```diff
@@ -8,35 +8,30 @@ import (
...
 
-	HostProc = "/host/proc"
-
 	unixDomainSocketDirectoryInContainer = "/host/var/lib/longhorn/unix-domain-socket/"
 )
 
 const (
 	BlockSizeLinux = 512
+
+	randomIDLenth = 8
```
Comment (innobead) [inline pkg/util/util.go:34]: `randomIDLenth` still used?

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Just a few comments.

File: pkg/util/util.go (hunk 5 above)
Comment (ChanYiLin) [reply pkg/util/util.go]: Fixed.

File: app/cmd/restore_to_file.go (hunk 3 above)
Comment (ChanYiLin) [reply app/cmd/restore_to_file.go]: Fixed.

**Comment by innobead** [comment]
Discussed with @ChanYiLin, all changes here are just from the master. If anything requires changes, let's fix them in the mater branch.

**Review by innobead — APPROVED** [review]

**Pull request merged**